internal/db/storage/blob/s3/s3_test.go

```diff
@@ -50,42 +53,91 @@ func TestNew(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+
+		if (w.want == nil && got != nil) || (w.want != nil && got == nil) {
+			return errors.Errorf("got: %v, want: %v", w.want, got)
+		} else if w.want == nil && got == nil {
+			return nil
 		}
+
+		wantC, gotC := w.want.(*client), got.(*client)
+
+		clientComparator := []comparator.Option{
+			comparator.AllowUnexported(*gotC),
+			comparator.Comparer(func(x, y *session.Session) bool {
+				return x != nil && y != nil
```

Comment:
should we use reflect.Equal here?

Reply:
we can not use reflect.DeepEqual here because session contains function type fields. so reflect.DeepEqual always returns false.

Reply:
or we can use gocmp to define the custom compartor for the function type